validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java

```diff
@@ -128,7 +132,14 @@ public void onPrimaryNodeBackReady() {
   }
 
   private BackgroundEventSource createEventSource(final RemoteValidatorApiChannel beaconNodeApi) {
-    final HttpUrl eventSourceUrl = createHeadEventSourceUrl(beaconNodeApi.getEndpoint());
+    final HttpUrl eventSourceUrl =
```

Comment:
How about this? More streamlined and easier to extend
```java
    final List<EventType> eventTypes = new ArrayList<>();
    eventTypes.add(EventType.head);
    if (validatorSlashingMonitoringEnabled) {
      eventTypes.add(EventType.attester_slashing);
      eventTypes.add(EventType.proposer_slashing);
    }
    final HttpUrl eventSourceUrl =
        createEventStreamSourceUrl(
            beaconNodeApi.getEndpoint(), eventTypes.toArray(EventType[]::new));
```

Reply:
Done. Updated the createEventStreamSourceUrl to take a list as argument...